internal/storage/bucket_handle.go

```diff
@@ -311,3 +311,47 @@ func (b *bucketHandle) UpdateObject(ctx context.Context, req *gcs.UpdateObjectRe
 
 	return
 }
+
+func (b *bucketHandle) ComposeObjects(ctx context.Context, req *gcs.ComposeObjectsRequest) (o *gcs.Object, err error) {
+	dstObj := b.bucket.Object(req.DstName)
+
+	if req.DstGenerationPrecondition != nil {
```

Comment:
what happends when DstGenerationPrecondition value is zero?

Reply:
it is composing object without any error

Reply:
for *dstMetageneration=0 it is throwing 
error  = Error in composing object: storage: ComposeFrom destination: empty conditions